simulation/navigator_gazebo/test/test_sim_integration.py

```diff
@@ -0,0 +1,205 @@
+                    data.pose.pose.position.z, beg_pos_z)))
+        self.assertEqual(
+            data.pose.pose.orientation.x, beg_orien_x, msg=(
+                "Error: /absodom "
+                "orientation x is: {} should be {}".format(
+                    data.pose.pose.orientation.x, beg_orien_x)))
+        self.assertEqual(
+            data.pose.pose.orientation.y, beg_orien_y, msg=(
+                "Error: /absodom "
+                "orientation y is: {} should be {}".format(
+                    data.pose.pose.orientation.y, beg_orien_y)))
+        self.assertEqual(
+            data.pose.pose.orientation.z, beg_orien_z, msg=(
+                "Error: /absodom "
+                "orientation z is: {} should be {}".format(
+                    data.pose.pose.orientation.z, beg_orien_z)))
+        self.assertEqual(
+            data.pose.pose.orientation.w, beg_orien_w, msg=(
+                "Error: /absodom "
+                "orientation w is: {} should be {}".format(
+                    data.pose.pose.orientation.w, beg_orien_w)))
+
+    def test_image(self):
+        # get camera information for all cameras
+        rospy.Subscriber(
+            "/camera/front/right/camera_info",
+            CameraInfo,
+            self.verify_image)
+        rospy.Subscriber(
+            "/camera/front/left/camera_info",
+            CameraInfo,
+            self.verify_image)
+        rospy.Subscriber(
+            "/camera/down/camera_info",
+            CameraInfo,
+            self.verify_image)
+        rospy.Subscriber(
+            "/camera/starboard/camera_info",
+            CameraInfo,
+            self.verify_image)
+        # get image information for all cameras
+        rospy.Subscriber(
+            "/camera/front/right/image_color",
+            Image,
+            self.verify_not_empty)
+        rospy.Subscriber(
+            "/camera/front/left/image_color",
+            Image,
+            self.verify_not_empty)
+        rospy.Subscriber(
+            "/camera/down/image_color",
+            Image,
+            self.verify_not_empty)
+        rospy.Subscriber(
+            "/camera/starboard/image_color",
+            Image,
+            self.verify_not_empty)
+
+    def verify_image(self, data):
+        res_height = 600
+        res_width = 960
+        self.assertEqual(
+            data.height, res_height, msg=(
+                "Error: /camera/front/right/camera_info "
+                "height is: {} should be {}".format(
+                    data.height, res_height)))
+        self.assertEqual(
+            data.width, res_width, msg=(
+                "Error: /camera/front/right/camera_info "
+                "width is: {} should be {}".format(
+                    data.width, res_width)))
+
+    def test_points(self):
+        rospy.Subscriber("/velodyne_points", PointCloud2, self.verify_points)
+        rospy.Subscriber("/velodyne_points", PointCloud2, self.verify_not_empty)
```

Comment:
Better to have one function verify_pointcloud that does both of these. You should never have multiple subscribers to the same topic